playbooks/tasks/portal-docker-services-start.yml

```diff
@@ -50,20 +50,30 @@
   debug:
     msg: "{{ inventory_hostname + ' docker services start: ' + lookup('pipe','date +%Y-%m-%dT%H:%M:%S') + ' UTC' }}"
 
-# Start the docker services (on deploy also build images)
-- name: Start the docker services (on deploy also build images)
-  community.docker.docker_compose:
-    project_src: "{{ webportal_dir }}"
-    files: "{{ webportal_docker_compose_files_wanted }}"
-    build: "{{ docker_compose_build }}"
-    remove_orphans: True
-    nocache: True
-    pull: True
-    state: present
-    # TODO: Build in parallel, when Ansible support is added
+- name: Pull docker images
+  ansible.builtin.command: "chdir={{ webportal_dir }} ./dc pull"
   become: True
-  become_user: "user"
-  register: docker_compose_start_result
+  become_user: "{{ webportal_user }}"
+
+- name: Build docker images
+  ansible.builtin.command: "chdir={{ webportal_dir }} ./dc build --no-cache"
```

Comment:
Am I reading this wrong or are we pulling and rebuilding the service with no cache each time we want to start it?

Reply:
Yes we are pulling images every time we start - images are cached though so its skipped if the versions are already pulled.

Yes we are rebuilding without cache every time we start but currently all our services are served from images so this step is skipped completely, it's only relevant when we override an image with link to Dockerfile and we want to make sure that is rebuilt without cache every time because the args for those are branch names and we want to pull the latest branch every time, otherwise it would get the stale version from cache - we're using that sometimes for debugging purposes.

Just to be clear, both these settings were already set up like that with ansible docker-compose plugin, this is merely transforming it into usage of dc instead.

Reply:
This is bad. We shouldn't be doing it. Not using the docker images is a valid approach and the portal operator shouldn't be punished by a massive delay each time they restart their stack. Let's remove the `--no-cache`, so at least we use the cache and don't incur the delay on each restart.

Reply:
We had this same exact discussion 3 months ago and the only way to redeploy latest changes if operator is targeting a branch is to use no-cache option, see https://github.com/SkynetLabs/ansible-playbooks/pull/328